src/TorchSharp/NN/Activation/PReLU.cs

```diff
@@ -15,33 +16,74 @@ namespace Modules
         /// </summary>
         public sealed class PReLU : torch.nn.Module<Tensor, Tensor>
         {
-            internal PReLU(IntPtr handle, IntPtr boxedHandle) : base(handle, boxedHandle) { }
+            internal PReLU(long num_parameters, double init) : base(nameof(PReLU)) 
+            { 
+                this._init = init;
+                // This will also set the weights
+                this.num_parameters = num_parameters;
+            }
 
             public override Tensor forward(Tensor tensor)
             {
-                var res = THSNN_PReLU_forward(handle, tensor.Handle);
-                if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                return new Tensor(res);
+                return torch.nn.functional.prelu(tensor, weight);
             }
 
             public override string GetName()
             {
                 return typeof(PReLU).Name;
             }
 
-            public Parameter? weight {
-                get {
-                    var res = THSNN_PReLU_weight(handle);
-                    if (res == IntPtr.Zero) { torch.CheckForErrors(); }
-                    return (res == IntPtr.Zero) ? null : new Parameter(res);
+            public Parameter weight {
+                get => _weight!;
+                set {
+                    if (value is null) throw new ArgumentNullException(nameof(weight));
+                    if (value.Handle != _weight?.Handle) {
+                        _weight?.Dispose();
+                        _weight = (value.DetachFromDisposeScope() as Parameter)!;
+                        ConditionallyRegisterParameter(nameof(weight), _weight);
+                    }
                 }
+            }
+
+            public long num_parameters {
+                get => _num_parameters;
                 set {
-                    if (value is null) throw new ArgumentNullException("weight cannot be set to 'null'");
-                    THSNN_PReLU_set_weight(handle, value!.Handle);
-                    torch.CheckForErrors();
-                    ConditionallyRegisterParameter("weight", value);
+                    if (value != _num_parameters)
+                    {
+                        this._num_parameters = value;
+                        var w = torch.empty(value);
+                        w.fill_(_init);
+                        this._weight = new Parameter(w);
+                    }
                 }
+
             }
+
+            public double init {
+                get => _init;
+                set {
+                    if (value != _init)
+                    {
+                        this._init = value;
+                        var w = torch.empty(_num_parameters);
+                        w.fill_(value);
+                        this._weight = new Parameter(w);
+                    }
+                }
+            }
```

Comment:
The behavior here isn't the same as in PyTorch. 
In PyTorch, you can change the constants without anything else updating. This is similar to our question about the `in_features` in the `Linear` module.

Reply:
Okay, that's just stupid. In that case, we should prevent those attributes from being modified.

Reply:
What about the `weight` attribute? 
We allow that to be modified, and so if that gets changed perhaps we should change the `in_features` property as well, no?

Reply:
I made the 'set' private.

Reply:
What about in `Linear`, we have the same situation there

Reply:
No, we can't make the weight setter private.

You've stayed up all night????

Reply:
> What about in `Linear`, we have the same situation there. It isn't necessarily correct to make the `weight` set private, since there is a plethora of usage in PyTorch where they modify the `weight` property directly.

We should definitely do the same for Linear and Bilinear.

Reply:
> You've stayed up all night????

Hahaha yup, we've got stuff to do!

> We should definitely do the same for Linear and Bilinear.

Sounds good!